### PR TITLE
Add warning about google-cloud-platform-core dependency change in #24235

### PR DESCRIPTION
**Pull request by apilloud**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by apilloud** [comment]
R: @lukecwik

File: CHANGES.md [hunk 1]
```diff
@@ -74,6 +74,10 @@
 
 * `ParquetIO.withSplit` was removed since splittable reading has been the default behavior since 2.35.0. The effect of
   this change is to drop support for non-splittable reading (Java)([#23832](https://github.com/apache/beam/issues/23832)).
+* `beam-sdks-java-extensions-google-cloud-platform-core` is no longer a
+  dependency of the Beam Portable SDK Harness. Some users of the portable runner
```
Comment (lukecwik) [inline CHANGES.md:78]:
```suggestion
  dependency of the Java SDK Harness. Some users of a portable runner
```

File: CHANGES.md [hunk 2]
```diff
@@ -74,6 +74,10 @@
...
+  may have an undeclared dependency on this package, for example using gcs with
+  TextIO, and will now need to declare the dependency.
```
Comment (lukecwik) [inline CHANGES.md:80]:
```suggestion
  may have an undeclared dependency on this package (for example using GCS with
  TextIO) and will now need to declare the dependency.
```

**Review by lukecwik — APPROVED** [review]

File: CHANGES.md (hunk 1 above)
Comment (apilloud) [reply CHANGES.md]: Done

File: CHANGES.md (hunk 2 above)
Comment (apilloud) [reply CHANGES.md]: Done.

**Pull request merged**